src/Symfony/Component/Security/Http/Firewall/SwitchUserListener.php

```diff
@@ -69,26 +71,32 @@ public function __construct(TokenStorageInterface $tokenStorage, UserProviderInt
      *
      * @param GetResponseEvent $event A GetResponseEvent instance
      *
-     * @throws \LogicException if switching to a user failed
+     * @throws \LogicException if switching to a user or exiting fails
      */
     public function handle(GetResponseEvent $event)
     {
         $request = $event->getRequest();
+        $usernameParameter = $request->get($this->usernameParameter);
 
-        if (!$request->get($this->usernameParameter)) {
+        // listener stops if _switch_user request parameters is not defined
+        if (!$usernameParameter) {
             return;
```

Comment:
Nope, then the whole request will be aborted. It goes straight to the
exception handlers, and no controllers will be called for any request
without the parameter.

--
Lars Vierbergen
On Jan 19, 2015 10:11 PM, "Emanuele Gaspari" <notifications@github.com>
wrote:

> In src/Symfony/Component/Security/Http/Firewall/SwitchUserListener.php
> <https://github.com/symfony/symfony/pull/12874#discussion-diff-23187110>:
>
> >
> > -        if (!$request->get($this->usernameParameter)) {
> > +        // listener stops if _switch_user request parameters is not defined
> > +        if (!$usernameParameter) {
> >              return;
>
> should this throw an exception?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/symfony/symfony/pull/12874/files#r23187110>.
>